src/util/permissions.ts

```diff
@@ -74,19 +74,19 @@ export async function botHasPermission(
       return bits;
     }, BigInt(0));
 
-  if (permissionBits && BigInt(Permissions.ADMINISTRATOR)) return true;
+  if (permissionBits & BigInt(Permissions.ADMINISTRATOR)) return true;
 
   return permissions.every((permission) =>
-    permissionBits && BigInt(Permissions[permission])
+    permissionBits & BigInt(Permissions[permission])
   );
 }
 
 /** Checks if the bot has the permissions in a channel */
-export function botHasChannelPermissions(
+export async function botHasChannelPermissions(
   channelID: string,
   permissions: Permission[],
 ) {
-  return hasChannelPermissions(channelID, botID, permissions);
+  return await hasChannelPermissions(channelID, botID, permissions);
```

Comment:
Since hasChannelPermissions() returns a Promise, you do not have to convert the function to `async`.

Reply:
It's seems to be needed the function doesn't work without it

Reply:
Confirmed, this does not seem to fix the issue either.